zerver/webhooks/slack/doc.md

```diff
@@ -1,34 +1,79 @@
-Get Zulip notifications from Slack for messages on your team's
-public channels!
+# Zulip Slack integration
 
-See also the [Slack-compatible webhook](/integrations/doc/slack_incoming).
+Get Zulip notifications for messages on your team's public Slack
+channels! You can choose to map each **Slack channel** either to a
+**Zulip topic** or to a **Zulip channel**.
 
-1. {!create-channel.md!}
+See also the [Zulip Slack incoming webhook integration][1].
+
+!!! warn ""
+
+    Using [Slack's legacy Outgoing Webhook service][5] is no longer
+    recommended. Follow these instructions to switch to the new
+    [Slack Events API][3].
+
+{start_tabs}
+
+1. To map each Slack channel to a Zulip topic, [create one channel][2]
+   you'd like to use for Slack notifications. Otherwise, for each public
+   Slack channel, [create a Zulip channel][2] with the same name.
 
 1. {!create-an-incoming-webhook.md!}
 
-1. {!generate-integration-url.md!}
+1. {!generate-webhook-url-basic.md!}
+
+    To map each Slack channel to a Zulip topic, make sure that the
+    **Send all notifications to a single topic** option is disabled
+    when generating the URL. Add `&channels_map_to_topics=1` to the
+    generated URL.
+
+    Otherwise, add `&channels_map_to_topics=0` to the generated URL.
+    Note that any Zulip channel you specified when generating the URL
+    will be ignored in this case.
+
+1. Create a new [Slack app][4], and open it. Navigate to the **OAuth
+   & Permissions** menu, and scroll down to the **Scopes** section.
+
+1. Make sure **Bot Token Scopes** includes `channels:read`,
+   `channels:history`, `users:read`, `emoji:read`, `team:read`,
+   `users:read`, and `users:read.email`.
```

Comment:
Do we need all of these scopes? I feel like we may want to add a section at the bottom of these docs explaining the reasons for the individual scopes for the paranoid, but maybe it's not worth the effort.

Reply:
Added this TODO to #30827 for tracking.

Reply:
yes since we need to do callbacks for the channel name and user name